WcaOnRails/app/jobs/submit_results_nag_job.rb

```diff
@@ -0,0 +1,14 @@
+class SubmitResultsNagJob < ActiveJob::Base
+  queue_as :default
+
+  def nag_needed(competition)
+    (competition.results_nag_sent_at || competition.end_date) <= 1.week.ago
+  end
+
+  def perform()
+    Competition.where(results_posted_at: nil).all.select { |c| nag_needed(c) }.each do |competition|
+      competition.update_attribute(:results_nag_sent_at, Time.now)
+      CompetitionsMailer.submit_results_nag(competition).deliver_later
```

Comment:
That makes sense. I guess I was thinking that after you set up a background job runner, you never use deliver_now, but as you said, it seems to make sense here. I've changed this to `deliver_now`.